indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java

```diff
@@ -1717,6 +1717,98 @@ public Response getUnparseableEvents(
     return Response.ok(parseExceptionHandler.getSavedParseExceptionReports()).build();
   }
 
+  @POST
+  @Path("/updateConfig")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response updateConfig(TaskConfigUpdateRequest<PartitionIdType, SequenceOffsetType> req) throws InterruptedException
+  {
+    try {
+      requestPause();
```

Comment:
We should call `pause()` instead of `requestPause()`, since the latter only requests a pause but doesn't ensure that we reach a paused state.
If the `pause()` call returns non-OK response, we should return the same response immediately.

Reply:
Before starting with pause, we can log the new config that we are trying to update to.
After the update finishes, let's log the new config and also emit an event.